src/aiida/engine/daemon/execmanager.py

```diff
@@ -201,96 +201,27 @@ def upload_calculation(
     remote_symlink_list = calc_info.remote_symlink_list or []
     provenance_exclude_list = calc_info.provenance_exclude_list or []
 
-    for uuid, filename, target in local_copy_list:
-        logger.debug(f'[submission of calculation {node.uuid}] copying local file/folder to {target}')
-
-        try:
-            data_node = load_node(uuid=uuid)
-        except exceptions.NotExistent:
-            data_node = _find_data_node(inputs, uuid) if inputs else None
-
-        if data_node is None:
-            logger.warning(f'failed to load Node<{uuid}> specified in the `local_copy_list`')
-        else:
-            # If no explicit source filename is defined, we assume the top-level directory
-            filename_source = filename or '.'
-            filename_target = target or ''
+    file_copy_order = calc_info.file_copy_order
 
-            # Make the target filepath absolute and create any intermediate directories if they don't yet exist
-            filepath_target = pathlib.Path(folder.abspath) / filename_target
-            filepath_target.parent.mkdir(parents=True, exist_ok=True)
-
-            if data_node.base.repository.get_object(filename_source).file_type == FileType.DIRECTORY:
-                # If the source object is a directory, we copy its entire contents
-                data_node.base.repository.copy_tree(filepath_target, filename_source)
-                sources = data_node.base.repository.list_object_names(filename_source)
-                if filename_target:
-                    sources = [str(pathlib.Path(filename_target) / subpath) for subpath in sources]
-                provenance_exclude_list.extend(sources)
-            else:
-                # Otherwise, simply copy the file
-                with folder.open(target, 'wb') as handle:
-                    with data_node.base.repository.open(filename, 'rb') as source:
-                        shutil.copyfileobj(source, handle)
-
-                provenance_exclude_list.append(target)
+    if file_copy_order is FileCopyOrder.REMOTE_LOCAL_SANDBOX:
+        if not dry_run:
+            _copy_remote_files(logger, node, computer, transport, remote_copy_list, remote_symlink_list)
+        provenance_exclude_list += _copy_local_files(logger, node, inputs, folder, local_copy_list)
+        if not dry_run:
+            _copy_sandbox_files(logger, node, transport, folder)
+    else:
```

Comment:
I would put here an `elif`, and raise in the `else` statement, so we don't forget to fix this if a new enum option is added in the future